src/torrent_handle.cpp

```diff
@@ -265,6 +265,29 @@ namespace libtorrent {
 		async_call(&torrent::pause, bool(flags & graceful_pause));
 	}
 
+	boost::uint64_t torrent_handle::flags() const
+	{
+		return sync_call_ret<boost::uint64_t>(0, &torrent::flags);
+	}
+
+	void torrent_handle::set_flags(boost::uint64_t flags, boost::uint64_t mask) const
+	{
+		void (torrent::*set_flags)(boost::uint64_t, boost::uint64_t) = &torrent::set_flags;
```

Comment:
it might be simpler if the internal type (``torrent``) only has the ``set_flags`` that takes a mask, and both of these call it. Then you don't need to cast to pick the right overload here